iphone/Classes/TiUIListView.m

```diff
@@ -2158,30 +2204,28 @@ - (void)presentSearchController:(UISearchController *)controller
   if (resultsSeparatorStyle) {
     [resultViewController.tableView setSeparatorStyle:[TiUtils intValue:resultsSeparatorStyle def:UITableViewCellSeparatorStyleSingleLine]];
   }
-
   // Presenting search controller on window holding controller
-  id proxy = [(TiViewProxy *)self.proxy parent];
-  while ([proxy isKindOfClass:[TiViewProxy class]] && ![proxy isKindOfClass:[TiWindowProxy class]]) {
-    proxy = [proxy parent];
+  if (!searchControllerPresenter) {
+    id proxy = [(TiViewProxy *)self.proxy parent];
+    while ([proxy isKindOfClass:[TiViewProxy class]] && ![proxy isKindOfClass:[TiWindowProxy class]]) {
+      proxy = [proxy parent];
+    }
+    if ([proxy isKindOfClass:[TiWindowProxy class]]) {
+      searchControllerPresenter = [proxy windowHoldingController];
+    } else {
+      searchControllerPresenter = [[TiApp app] controller];
```

Comment:
Did you ensure this does not get released before closing the window?

Reply:
This is not needed it is just used to push the controller and every time before presenting controller it is calculated again.